components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/RoleManagementService.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.role.mgt.core;
+
+import java.util.List;
+
+/**
+ * OSGi service interface which use to manage roles.
+ */
+public interface RoleManagementService {
+
+    /**
+     * Add a new role.
+     *
+     * @param roleName     Role name.
+     * @param userList     List of users.
+     * @param groupList    List of groups.
+     * @param permissions  List of permissions.
+     * @param tenantDomain Tenant domain.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    void addRole(String roleName, List<String> userList, List<String> groupList, List<String> permissions,
```

Comment:
shouldn't we pass userstore domain here?

Reply:
According to the new concept, roles are stored internally. Only groups are specified under user stores.